sdk/src/prover.rs

```diff
@@ -0,0 +1,164 @@
+
+        // Prepare input data and upload it.
+        let input_id = { client.upload_input(input)? };
+
+        // Start a session running the prover.
+        let session = client.create_session(image_id_hex, input_id, vec![])?;
+        log::info!("Created session: {}", session.uuid);
+        let _receipt = loop {
+            let res = session.status(&client)?;
+            if res.status == "RUNNING" {
+                log::info!(
+                    "Current status: {} - state: {} - continue polling...",
+                    res.status,
+                    res.state.unwrap_or_default()
+                );
+                std::thread::sleep(Duration::from_secs(15));
+                continue;
+            }
+            if res.status == "SUCCEEDED" {
+                // Download the receipt, containing the output.
+                let receipt_url = res
+                    .receipt_url
+                    .expect("API error, missing receipt on completed session");
+
+                let receipt_buf = client.download(&receipt_url)?;
+                let receipt: Receipt = bincode::deserialize(&receipt_buf)?;
+
+                break receipt;
+            } else {
+                panic!(
+                    "Workflow exited: {} - | err: {}",
+                    res.status,
+                    res.error_msg.unwrap_or_default()
+                );
+            }
+        };
+
+        // Fetch the snark.
+        let snark_session = client.create_snark(session.uuid)?;
+        log::info!("Created snark session: {}", snark_session.uuid);
+        let snark_receipt = loop {
+            let res = snark_session.status(&client)?;
+            match res.status.as_str() {
+                "RUNNING" => {
+                    log::info!("Current status: {} - continue polling...", res.status,);
+                    std::thread::sleep(Duration::from_secs(15));
+                    continue;
+                }
+                "SUCCEEDED" => {
+                    break res.output.expect("No snark generated :(");
+                }
+                _ => {
+                    panic!(
+                        "Workflow exited: {} err: {}",
+                        res.status,
+                        res.error_msg.unwrap_or_default()
+                    );
+                }
+            }
+        };
+
+        let snark = snark_receipt.snark;
+        log::debug!("Snark proof!: {snark:?}");
+
+        // // Verify receipt.
+        // let receipt = Receipt {
+        //     inner: risc0_zkvm::InnerReceipt::Groth16(Groth16Receipt {
+        //         seal: Groth16Seal {
+        //             a: snark.a.clone(),
+        //             b: snark.b.clone(),
+        //             c: snark.c.clone(),
+        //         }
+        //         .to_vec(),
+        //         claim: receipt.get_claim()?,
+        //     }),
+        //     journal: receipt.journal,
+        // };
+        // receipt.verify(image_id)?;
```

Comment:
This is commented only because we have a different (rust) verifying key on risc0 at the moment. Should be fixed for 0.21